### DISCUSSION TITLE
Avoid jumping into NvimTree buffer

### DISCUSSION THREAD

**Discussion by n3wborn** [cbochs/portal.nvim#26] [Q&A] [answered]
Hi ! Thanks for the time you spend working on portal and sharing this with the community.

I was wondering...

- I'm a big fan of jumplist
- I use NvimTree and have an autocommand that leave nvim when NvimTree is the last opened buffer
- Once I restart nvim, if I leaved it using the autocommand and do jump backward, nvim closes again (after all, that's what I asked with the autocommand)

So my question is: 

When jumping forward/backward, is there a way to filter/exclude jumps related to NvimTree filetype/bufname/etc ?

I thought I could do this using builtin and filter but it seems I don't know what I'm doing ^^

**Comment by cbochs** [comment, marked as answer]
Sure thing. You _should_ be able to filter jumps by filtype:

**During setup**

Apply to all searches (edited for v0.5.0 release).

```lua
require("portal").setup({
  filter = function(v)
    return vim.api.nvim_buf_get_option(v.buffer, "filetype") ~= "{ft}"
  end
})
```

**Function call**

Apply to a single search.

```lua
require("portal.builtin").jumplist.tunnel_backward({
  filter = function(v)
    return vim.api.nvim_buf_get_option(v.buffer, "filetype") ~= "{ft}"
  end
})
```

I'm not certain what the filetype for nvim-tree so I've left that as `{ft}` in the examples above. If should be listed in their docs, or you can simply run the following command while focused on an nvim-tree buffer:

```
:lua =vim.api.nvim_buf_get_option(0, "filetype")
```

**Reply by n3wborn** [reply]
Thanks a lot :)

**Reply by cbochs** [reply]
@n3wborn Edited the above **During setup** code for the recent [v0.5.0](https://github.com/cbochs/portal.nvim/releases/tag/v0.5.0) release
